### PR TITLE
Adopt Zatoshi object from SDK 1.7

### PR DESCRIPTION
**Pull request by ccjernigan**
This is a sort of quick and dirty adoption of the Zatoshi API.

A lot of the code in this app is based on hardcoded special values for Zatoshi (e.g. -1 being uninitialized).  With the new Zatoshi API, uninitialized values should be `null`.  Rather than doing a deep refactoring, you'll see some slightly gross conversions.

Example: 
`zatoshiAmount < 1` becomes `zatoshiAmount?.let { it.value < 1L } ?: false`

Since we're implementing the new app project, I think these in-place conversions are a reasonable tradeoff between updating this project to adopt the new SDK versus spending a lot of time trying to deeply change what's already here.

### CODE REVIEW

**Comment by ccjernigan** [comment]
> @ccjernigan @pacu please guys, can I get write access to this repo? I've just wanted to push a small change, but I couldn't. Thank you!

I made the internal IT request for this.

File: app/src/main/java/cash/z/ecc/android/ui/send/SendViewModel.kt [hunk 1]
```diff
@@ -86,7 +87,7 @@ class SendViewModel @Inject constructor() : ViewModel() {
         reportUserInputIssues(memoToSend)
         return synchronizer.sendToAddress(
             keys[0],
-            zatoshiAmount,
+            zatoshiAmount!!,
```
Comment (pacu) [inline app/src/main/java/cash/z/ecc/android/ui/send/SendViewModel.kt:90]: Will this crash if the value is null?

**Review by pacu — APPROVED** [review]
LGTM! just left a comment on the send flow code

**Review by HonzaR — APPROVED** [review]
Thanks Carter, the problem with the rights was again in the my git client setting, but the client just said that I haven't the required rights 😅. In the meantime you pushed the same changes, which I was about to push yesterday (WalletBalance imports), so I found another minor change to check the changed client settings. Now it works! Other than that, it seems alright to me.

**Pull request merged**